jax/interpreters/partial_eval.py

```diff
@@ -1166,7 +1165,9 @@ def ensure_instantiated(inst: bool, x: Atom) -> Atom:
         inputs = map(ensure_instantiated, inst_in, eqn.invars)
         staged_eqns.append(eqn.replace(invars=inputs))
         map(partial(write, False, True), eqn.outvars)
-  out_unknowns, out_inst = unsafe_map(list, unzip2(map(read, jaxpr.outvars)))
+  out_unknowns: Sequence[bool]
+  out_inst: Sequence[bool]
+  out_unknowns, out_inst = unzip2(map(read, jaxpr.outvars))
```

Comment:
I'm not a big fan of declaring variables before they're bound. As with `cast`, it seems like just making Python more verbose.

If there's no way to annotate this on a single line, can we `# type: ignore` instead?

Reply:
I changed it to not use declared variables, but I couldn't figure out how to do it in one line.